design/repository-maintenance.md

```diff
@@ -0,0 +1,236 @@
+		cli,
+		f.Namespace(),
+		"/tmp/credentials",
+		filesystem.NewFileSystem(),
+	)
+	cmd.CheckError(err)
+	credentialSecretStore, err := credentials.NewNamespacedSecretStore(cli, f.Namespace())
+	cmd.CheckError(err)
+	// Initialize repo provider
+	repoProvider := provider.NewUnifiedRepoProvider(
+		credentials.CredentialGetter{
+			FromFile:   credentialFileStore,
+			FromSecret: credentialSecretStore,
+		}, o.RepoType, cli, logrus.New())
+	
+	// Get backupRepository
+	repo, err := repository.GetBackupRepository(context.Background(), cli, f.Namespace(),
+		repository.BackupRepositoryKey{
+			VolumeNamespace: o.VolumeNamespace,
+			BackupLocation:  o.BackupStorageLocation,
+			RepositoryType:  o.RepoType,
+		}, true)
+	cmd.CheckError(err)
+	// Get BSL
+	bsl := &velerov1api.BackupStorageLocation{}
+	cli.Get(context.Background(), client.ObjectKey{Namespace: f.Namespace(), Name: repo.Spec.BackupStorageLocation}, bsl)
+	cmd.CheckError(err)
+
+	para := provider.RepoParam{
+		BackupRepo:     repo,
+		BackupLocation: bsl,
+	}
+	// Connect
+	err = repoProvider.BoostRepoConnect(context.Background(), para)
+	cmd.CheckError(err)
+	// Prune repo
+	err = repoProvider.PruneRepo(context.Background(), para)
+	cmd.CheckError(err)
+	...
+}
+```
+
+### 2. Create Jobs by Repository Manager
+Currently, the backup repository controller will call the repository manager to do the `PruneRepo`, and Kopia or Restic maintenance is then finally called through multiple chain calls.
+
+We will keep using the `PruneRepo` function in the repository manager, but we cut off the multiple chain calls by creating a maintenance job.
+
+```golang
+func (m *manager) PruneRepo(repo *velerov1api.BackupRepository) error {
+	...
+	// writing the maintenance result into default terminationMessagePath
+	args := []string {
+	"-c",
+	"velero maintenance --repo-name=" + param.BackupRepo.Spec.VolumeNamespace + " --repo-type=" + param.BackupRepo.Spec.RepositoryType + " --backup-storage-location=" + param.BackupLocation.Name + " > /dev/termination-log",
+	}
+
+	env := &v1.EnvVar{Name: ProviderFile, Value: "/credentials/cloud"}
+	job := builder.ForJob(param.BackupRepo.Name, param.BackupRepo.Namespace, "velero/velero:maintenance", args).Env(env).
+		VolumeMounts(*builder.ForVolumeMount("cloud-credentials", "/credentials").Result()).
+		ServiceAccount("velero").Resources(&resources).
+		Volumes(*builder.ForVolume("cloud-credentials").SecretSource("cloud-credentials").Result()).Result()
+	...
+	// Maintenance job also inherits some other filed values from the Velero server pod in keeping with the setting of the Velero server pod
+	log.Debugf("Creating maintenance job: %v", job)
+	defer cancel()
+	if err := m.client.Create(pruneTimeOut, job); err != nil {
+		return errors.Wrap(err, "error to create maintenance job")
+	}
+
+	if err := builder.WaitForJobComplete(pruneTimeOut, m.client, job); err != nil {
+		return errors.Wrap(err, "error to wait for maintenance job complete")
+	}
+	...
```

Comment:
Let's get the job result from here, since the job is launched by repoManager, the repoManager should get the result from the job, as the controller doesn't know anything about the job.